src/components/common/dashboard/OperationComponent.vue

```diff
@@ -0,0 +1,386 @@
+            const operation = ref(props.operation as Operation);
+            const approvals = operation.value.existingApprovals.users.length + operation.value.existingApprovals.groups.length;
+            const neededApprovals =
+                approvals + operation.value.missingApprovals.users.length + operation.value.missingApprovals.groups.length;
+            const missingApprovals = neededApprovals - approvals;
+
+            const isRejected = operation.value.state === OperationStatus.REJECTED;
+            const isPending = operation.value.state === OperationStatus.PENDING;
+            const isFinished = operation.value.state === OperationStatus.FINISHED || isRejected;
+            const actionRequired =
+                (operation.value.missingApprovals.users.includes(props.enrollmentID) ||
+                    operation.value.missingApprovals.groups.includes(props.role)) &&
+                !isRejected;
+
+            const sentApprove = ref(store.getters.treatedOperations.approved.includes(operation.value.operationId));
+            const sentReject = ref(store.getters.treatedOperations.rejected.includes(operation.value.operationId));
+            const provideReason = ref(false);
+            const selectedReason = ref("");
+            const writtenReason = ref("");
+            const isMyOperation = operation.value.initiator == props.enrollmentId;
+            const showWatchOption = !isMyOperation && isPending;
+
+            const isAdmin = props.role == Role.ADMIN;
+            const username = ref("...");
+
+            const isWatched = ref(props.watched);
+
+            const dateFormatOptions = {
+                weekday: "short",
+                year: "numeric",
+                month: "numeric",
+                day: "numeric",
+                hour: "2-digit",
+                minute: "2-digit",
+            };
+
+            const params = ref([] as string[]);
+            const title = ref("");
+            const shownOpId = operation.value.operationId.substring(0, 4);
+
+            const initiatedTimestamp = new Date(operation.value.initiatedTimestamp).toLocaleString("en-US", dateFormatOptions);
+            const lastUpdateTimestamp = new Date(operation.value.lastModifiedTimestamp).toLocaleString("en-US", dateFormatOptions);
+
+            const showDetails = ref(false);
+
+            onBeforeMount(async () => {
+                if (isAdmin) {
+                    getNameByEnrollmentId();
+                }
+                createDisplayObjects();
+            });
+
+            const type = computed(() => {
+                switch (operation.value.transactionInfo.contractName) {
+                    case UC4Identifier.CONTRACT_MATRICULATION:
+                        return "Matriculation";
+                    case UC4Identifier.CONTRACT_ADMISSION:
+                        return "Course Admission";
+                    default:
+                        return "";
+                }
+            });
```

Comment:
could be moved to OperationPrinter
maybe some "getOperationBadgeIdentifier()) or something

Reply:
Changed. I additionally move `OperationPrinter.ts` to `@/use/helpers` and updated the imports